File: testfiles/prime.rb
def find_first_prime(numbers)
  for n in numbers
    if n < 2
      next
    end
    is_prime = true
    d = 2
    while d * d <= n
      if n % d == 0
        is_prime = false
        break
      end
      d = d + 1
    end
    if is_prime
      return n
    end
  end
  return -1
end

File: testfiles/prime.ts
interface Result {
  value: number;
  found: boolean;
}

function findFirstPrime(numbers: number[]): number | null {
  for (let i = 0; i < numbers.length; i++) {
    const n: number = numbers[i];
    if (n < 2) {
      continue;
    }
    let isPrime: boolean = true;
    let d: number = 2;
    while (d * d <= n) {
      if (n % d === 0) {
        isPrime = false;
        break;
      }
      d = d + 1;
    }
    if (isPrime) {
      return n;
    }
  }
  return null;
}
